pkg/mixin/provider/install.go

```diff
@@ -15,11 +17,73 @@ import (
 )
 
 func (fs *FileSystem) Install(opts mixin.InstallOptions) (*mixin.Metadata, error) {
+	var err error
+	var metadata *mixin.Metadata
 	if opts.FeedURL != "" {
-		return fs.InstallFromFeedURL(opts)
+		metadata, err = fs.InstallFromFeedURL(opts)
+	} else {
+		metadata, err = fs.InstallFromURL(opts)
 	}
+	if err != nil {
+		return nil, err
+	}
+	fs.saveMixinInfo(opts)
+	return metadata, nil
+}
+
+const MixinCacheJSON string = "cache.json"
+
+func (fs *FileSystem) saveMixinInfo(opts mixin.InstallOptions) error {
+	mixinsDir, err := fs.Config.GetMixinsDir()
```

Comment:
Here you can use `mixinsDir, _` since we are ignoring the error returned by `GetMixinsDir`. Since at this point we are quite sure that method will succeed, it's okay to just ignore the error. 

I have had it on my todo list for quite a while to refactor all of these methods that get a directory relative to the porter home directory to not return an error. Now my inaction has come back to haunt others too! 👻 Sorry about that.

Reply:
No problem! Maybe I can help clean things up in the near future!